### DISCUSSION TITLE
How to check if txn is failed or succeed??

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#373] [Questions] [answered]
how do i check if txn failed or succeed?? `` const payload: Types.EntryFunctionPayload = { function: `${ABI.address}::some::author`, type_arguments: [], arguments: ["1"], }; const rawTxn = await this.client.generateTransaction(account.address(), payload); const signedTransaction = await this.client.signTransaction(account, rawTxn); const transactionResult = await this.client.submitTransaction(signedTransaction); await this.client.waitForTransaction(transactionResult.hash); `` for me this txn is failing but want an error thrown with the cause of txn failure

**Comment by 0xmaayan** [comment, marked as answer]
```
const response = await this.client.waitForTransactionWithResult(transactionResult.hash);
```
This returns the transaction details with the reason of failure

A side note: I suspect you are using the legacy TS SDK version - you might want to start using the new version https://aptos.dev/en/build/sdks/ts-sdk

**Comment by ajaythxkur** [comment]
Alright, how can i get account with this lib
`const privateKeyBytes = HexString.ensure(APTOS_SECRET_KEY).toUint8Array(); // Private key is hex string
`
and sign and submit txns?? 

Here - Account.fromPrivateKey({ privateKey: ??}),

**Reply by ajaythxkur** [reply]
Got it
`Account.fromPrivateKey({ privateKey: new Ed25519PrivateKey(APTOS_SECRET_KEY) })
`
